src/Facets/OpBNBBridgeFacet.sol

```diff
@@ -0,0 +1,196 @@
+import { IL1StandardBridge } from "../Interfaces/IL1StandardBridge.sol";
+import { LibAsset, IERC20 } from "../Libraries/LibAsset.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { InvalidConfig, AlreadyInitialized, NotInitialized } from "../Errors/GenericErrors.sol";
+import { SwapperV2, LibSwap } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { LibUtil } from "../Libraries/LibUtil.sol";
+
+/// @title OpBNB Bridge Facet
+/// @author Li.Finance (https://li.finance)
+/// @notice Provides functionality for bridging through OpBNB Bridge
+/// @custom:version 1.0.0
+contract OpBNBBridgeFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// Storage ///
+
+    bytes32 internal constant NAMESPACE = keccak256("com.lifi.facets.opbnb");
+
+    /// Types ///
+
+    struct Storage {
+        mapping(address => IL1StandardBridge) bridges;
+        IL1StandardBridge standardBridge;
+    }
+
+    struct Config {
+        address assetId;
+        address bridge;
+    }
+
+    struct OpBNBData {
+        address assetIdOnL2;
+        uint32 l2Gas;
+        bool isSynthetix;
+    }
+
+    /// Events ///
+
+    event OptimismInitialized(Config[] configs);
+    event OptimismBridgeRegistered(address indexed assetId, address bridge);
+
+    /// Init ///
+
+    /// @notice Initialize local variables for the OpBNB Bridge Facet
+    /// @param configs Bridge configuration data
+    function initOpBNB(
+        Config[] calldata configs,
+        IL1StandardBridge standardBridge
+    ) external {
+        LibDiamond.enforceIsContractOwner();
+
+        Storage storage s = getStorage();
+
+        for (uint256 i = 0; i < configs.length; i++) {
+            if (configs[i].bridge == address(0)) {
+                revert InvalidConfig();
+            }
+            s.bridges[configs[i].assetId] = IL1StandardBridge(
+                configs[i].bridge
+            );
+        }
+
+        s.standardBridge = standardBridge;
+
+        emit OptimismInitialized(configs);
+    }
+
+    /// External Methods ///
+
+    /// @notice Register token and bridge
+    /// @param assetId Address of token
+    /// @param bridge Address of bridge for asset
+    function registerOpBNBBridge(address assetId, address bridge) external {
+        LibDiamond.enforceIsContractOwner();
+
+        Storage storage s = getStorage();
+
+        if (bridge == address(0)) {
+            revert InvalidConfig();
+        }
+
+        s.bridges[assetId] = IL1StandardBridge(bridge);
+
+        emit OptimismBridgeRegistered(assetId, bridge);
+    }
+
+    /// @notice Bridges tokens via OpBNB Bridge
+    /// @param _bridgeData Data contaning core information for bridging
+    /// @param _bridgeData Data specific to OpBNB Bridge
+    function startBridgeTokensViaOpBNBBridge(
+        ILiFi.BridgeData memory _bridgeData,
```

Comment:
this can be calldata (even if you call _startBridge which requires memory), it works (as used in OftWrapperFacet